wagtail/api/v2/views.py

```diff
@@ -336,10 +337,16 @@ def get_urlpatterns(cls):
         """
         This returns a list of URL patterns for the endpoint
         """
+        if WAGTAIL_APPEND_SLASH:
+            detail_path = '<int:pk>/'
+            find_path = 'find/'
+        else:
+            detail_path = '/<int:pk>'
+            find_path = '/find'
```

Comment:
The Django's `APPEND_SLASH=True` setting is not intended to force all urls to end with a slash. The CommonMiddleware docs explain it well:

> If APPEND_SLASH is True and the initial URL doesn’t end with a slash, and it is not found in the URLconf, then a new URL is formed by appending a slash at the end. If this new URL is found in the URLconf, then Django redirects the request to this new URL. Otherwise, the initial URL is processed as usual.
> For example, foo.com/bar will be redirected to foo.com/bar/ if you don’t have a valid URL pattern for foo.com/bar but do have a valid pattern for foo.com/bar/.
> https://docs.djangoproject.com/en/3.2/ref/middleware/#django.middleware.common.CommonMiddleware

`WAGTAIL_APPEND_SLASH` should behave similar to Django’s `APPEND_SLASH`. That is: if a url pattern without trailing slash exists and it matches the requested url, just serve it.

Reply:
I guess the code doesn't make it clear, but that should already be true. Requesting the API views that I created in my Wagtail project with or without an end-slash both work. That _may_ be due to a custom middleware that we use, but I'm not sure.